applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -340,8 +376,29 @@ static void device_status(enum usb_dc_status_code cb_status, const u8_t *param)
 
 static void protocol_change(u8_t protocol)
 {
+	BUILD_ASSERT_MSG(IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_DISABLED) ==
+			 !IS_ENABLED(CONFIG_USB_HID_BOOT_PROTOCOL),
+			 "Boot protocol setup inconsistency");
+	BUILD_ASSERT_MSG(IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_KEYBOARD) ==
+			 (IS_ENABLED(CONFIG_USB_HID_BOOT_PROTOCOL) && (CONFIG_USB_HID_PROTOCOL_CODE == 1)),
+			 "Boot protocol code does not reflect selected interface");
+	BUILD_ASSERT_MSG(IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_MOUSE) ==
+			 (IS_ENABLED(CONFIG_USB_HID_BOOT_PROTOCOL) && (CONFIG_USB_HID_PROTOCOL_CODE == 2)),
+			 "Boot protocol code does not reflect selected interface");
+
+	__ASSERT_NO_MSG((protocol == HID_PROTOCOL_BOOT) ||
+			(protocol == HID_PROTOCOL_REPORT));
+
+	if (IS_ENABLED(CONFIG_DESKTOP_HID_BOOT_INTERFACE_DISABLED)) {
+		LOG_WRN("BOOT protocol is not supported");
```

Comment:
Fixed, also added a proper check for a release build.
what do you mean by 

> We should also assert here to catch the unexpected callback call in debug configurations.

Reply:
The boot protocol should never be requested when boot interface is disabled. We can assert instead of just logging warning to make sure that this never happens.

Reply:
This can happen as data comes from the outside. I added a check and return if boo protocol is not supported.

Reply:
You are right.